### PR TITLE
User-ID

### LINKED ISSUE #3166: [New addon]: Show user id in profiles
### Is your request related to a problem? Please describe.

I don't want to go to the API and read raw json always
### Describe the solution you'd like

An addon where it shows the user id of the user on a profile, presumably next to the country of the user, moving the my-ocular status to the right

### Describe alternatives you've considered

None

### PR DESCRIPTION
**Pull request by CodeGuy92**
Resolves #3166

### Changes

I left comments.
All i need now is versionAdded, Credits, and some other stuff for addon.json
image

### Reason for changes

https://github.com/ScratchAddons/ScratchAddons/issues/3166

### Tests

https://github.com/ScratchAddons/ScratchAddons/issues/3166#issuecomment-890314338
To test this locally, click this: https://github.com/CodeGuy92/ScratchAddons/archive/refs/heads/CodeGuy92-user.zip

### CODE REVIEW

**Comment by FunctionalMetatable** [comment]
Resolves #3166

**Comment by CodeGuy92** [comment]
> Resolves #3166 <- put this in pr post

Ok

File: addons/user-id/addon.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+  "description": "Allows you to view any user's ID on their profile page.",
+  "userscripts": [
+    {
+      "url": "userscript.js",
+      "matches": ["https://scratch.mit.edu/users/*"]
+    }
+  ],
+  "tags": ["community", "profiles"],
+  "enabledByDefault": false
+}
```
Comment (FunctionalMetatable) [inline addons/user-id/addon.json:12]:
```suggestion
{
  "name": "Show user IDs",
  "description": "Allows you to view any user's ID on their profile page.",
  "credits": [
  	{
  		"name": "LankyBox01",
  		"link": "https://lankybox01.leahcimto.com/"
    }
  ],
  "userscripts": [
    {
      "url": "userscript.js",
      "matches": ["https://scratch.mit.edu/users/*"]
    }
  ],
  "tags": ["community", "profiles"],
  "versionAdded": "1.19.0",
  "enabledByDefault": false
}
```

File: addons/user-id/userscript.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+// fetch the user's id by making a get request to api.scratch.mit.edu
+// location.pathname returns /users/USERNAME/
+fetch('https://api.scratch.mit.edu' + location.pathname)
+  .then(response => response.json())
+  // Add the ID of the response to the username header
+  .then(data => userid.textContent = "#" + data.id);
+
+// add the id next to the header
+usernameheader.appendChild(userid);
```
Comment (FunctionalMetatable) [inline addons/user-id/userscript.js:17]:
```suggestion
export default async function ({ addon }) {
	let usernameHeader = document.querySelector("header-text").firstElementChild;
	const userId = Object.assign(document.createElement("font"), {
		innerText: `#${Scratch.INIT_DATA.PROFILE.model.userId}`,
		className: "sa-user-id"
	});
	userId.style.color = "lightgrey";
	userId.style.size = 2;
	
	usernameHeader.appendChild(userId)
```

Comment (CodeGuy92) [reply addons/user-id/userscript.js:17]: why

Comment (FunctionalMetatable) [reply addons/user-id/userscript.js:17]: it doesnt include requests, it uses data built in to scratch's profile page

File: addons/user-id/userscript.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export default async function ({ addon, global, console }) {
+// define the user's username header
+let usernameheader = document.getElementsByClassName("header-text")[0].firstElementChild
+
+let userid = document.createElement("font");
```
Comment (RedGuy12) [inline addons/user-id/userscript.js:5]: Font is deprecated

**Comment by RedGuy12** [comment]
Enable actions at https://github.com/CodeGuy92/ScratchAddons/actions

File: addons/user-id/userscript.js (hunk 2 above)
Comment (CodeGuy92) [reply addons/user-id/userscript.js]: Uncaught (in promise) TypeError: Cannot read property 'firstElementChild' of null

File: addons/user-id/userscript.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 export default function ({ addon }) {
 	let usernameHeader = document.querySelector("header-text").firstElementChild;
-	const userId = Object.assign(document.createElement("font"), {
+	const userId = Object.assign(document.createElement("["), {
```
Comment (RedGuy12) [inline addons/user-id/userscript.js:3]: Just use a span!

Comment (CodeGuy92) [reply addons/user-id/userscript.js:3]: why was it set to ['
uhh

File: addons/user-id/userscript.js (hunk 2 above)
Comment (FunctionalMetatable) [reply addons/user-id/userscript.js]: oops its `document.querySelector(".header-text").firstElementChild`

Comment (CodeGuy92) [reply addons/user-id/userscript.js]: how is that different

File: addons/user-id/userscript.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+export default function ({ addon }) {
+  let usernameHeader = document.querySelector("header-text").firstElementChild;
```
Comment (FunctionalMetatable) [inline addons/user-id/userscript.js:2]:
```suggestion
  let usernameHeader = document.querySelector(".header-text").firstElementChild;
```

**Comment by CodeGuy92** [comment]
Now it works, tests are still up-to-date

**Comment by CodeGuy92** [comment]
image

File: addons/user-id/userscript.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+export default function ({ addon }) {
+  let usernameHeader = document.querySelector(".header-text").firstElementChild;
+  const userId = Object.assign(document.createElement("span"), {
+    innerText: `#${Scratch.INIT_DATA.PROFILE.model.userId}`,
+    className: "sa-user-id",
+  });
+  userId.style.color = "lightgrey";
+  userId.style.size = 2;
```
Comment (RedGuy12) [inline addons/user-id/userscript.js:8]: `"2px"`

**Review by FunctionalMetatable — APPROVED** [review]
image
LGTM

File: addons/user-id/userscript.js (hunk 6 above)
Comment (CodeGuy92) [reply addons/user-id/userscript.js:8]: No, that would make it the size of a rice grain. The "2" is set correctly.

Comment (CodeGuy92) [reply addons/user-id/userscript.js:8]: Correction: Smaller than the size of a rice grain

Comment (RedGuy12) [reply addons/user-id/userscript.js]: O right

**Comment by jeffalo** [comment]
can the text be smaller?

**Comment by WorldLanguages** [comment]
We probably need a "minor addons" category by now :P

**Comment by CodeGuy92** [comment]
> can the text be smaller?

yea sure



> We probably need a "minor addons" category by now :P

lool

**Comment by CodeGuy92** [comment]
image

Updated tests :)

**Comment by CodeGuy92** [comment]
127749724-85e51b98-948d-4649-8508-18899ae38155

**Comment by WorldLanguages** [comment]
Waiting for #3261 to be merged

**Comment by CodeGuy92** [comment]
> Waiting for #3261 to be merged

aight

**Pull request merged**